packages/near-membrane-base/src/membrane.ts

```diff
@@ -879,12 +895,7 @@ export function createMembraneMarshall() {
             }
 
             private makeProxyUnambiguous(shadowTarget: ShadowTarget) {
-                if (
-                    foreignCallableHasOwnProperty(
-                        this.foreignTargetPointer,
-                        LOCKER_LIVE_MARKER_SYMBOL
-                    )
-                ) {
+                if (this.isArray || isPlainObjectOrEquivalent(this.foreignTargetPointer)) {
```

Comment:
remember that this is the slow path anyways... so checking for the traits first which is faster, then checking for the proto chain, and finally for the live marker seems like the right order.